plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/commit/CommitPresenter.java

```diff
@@ -236,10 +245,33 @@ private void push(Path location) {
             })
         .catchError(
             error -> {
-              notificationManager.notify(constant.pushFail(), FAIL, FLOAT_MODE);
+                final String providerName = GitUtil.getProviderNameFromError(error);
+                if (!isNullOrEmpty(providerName)) {
+                  pushAuthenticated(location, branch, remote, providerName);
+                  return;
+                }
+                notificationManager.notify(constant.pushFail(), FAIL, FLOAT_MODE);
             });
   }
 
+  protected void pushAuthenticated(
```

Comment:
Please consider a way to get rid of extra method in all presenters by extending a common GitPresenter. This is already done for SVN in master branch: https://github.com/eclipse/che/blob/5b373e6cf5bfeb16f278c4d750fe64361445eeb4/plugins/plugin-svn/che-plugin-svn-ext-ide/src/main/java/org/eclipse/che/plugin/svn/ide/common/SubversionActionPresenter.java#L138-L175